tests/TestCase/Routing/RouterTest.php

```diff
@@ -2260,6 +2260,24 @@ public function testParseRoutePath(): void
             'action' => 'view',
         ];
         $this->assertSame($expected, Router::parseRoutePath('Vendor/Cms.Management/Admin/Articles::view'));
+
+        $expected = [
+            'controller' => 'Bookmarks',
+            'action' => 'view',
+            '1',
+        ];
+        $this->assertSame($expected, Router::parseRoutePath('Bookmarks::view/1'));
+        $this->assertSame($expected, Router::parseRoutePath('Bookmarks::view/1/'));
+
+        $expected = [
```

Comment:
Such an array would only be useful for route ending with `/*` or `/**`, since for named route elements the element name is required as key. I am not sure we should be providing feature which only work for a subset of routes.

Other alternative would be for e.g.  supporting string like `Cms.Articles::edit/year:2023/month:03/day:5th` which gets parsed into array like:
```
[
    'plugin' => 'Cms',
    'controller' => 'Articles',
    'action' => 'edit',
    'year' => '2023',
    'month' => '03',
    'day' => '5th',
]
```

Reply:
how would this named arguments approach work with something like the PagesController::display method?
I e.g. have a Page URL which looks like `pages/statistic/2023/03` and therefore would require something like 
```
Pages::display/2023/03
```

Reply:
Routes that don't have named element won't use any :)

Reply:
I haven't gone through the code (or actually used it) but Croogo has an implementation of what I am suggesting:
https://github.com/croogo/croogo/blob/fc0648659dcb5790ba8a7429250dae492ca724ec/Core/src/Utility/StringConverter.php#L97

Reply:
I have adapted the regex to support "named arguments" kind of strings even though I am not sure which other string parts can be parts of the "path"

Reply:
> Another alternative would be for e.g. supporting string like Cms.Articles::edit/year:2023/month:03/day:5th which gets parsed into array like:

Would we have any rules on what order parameters of different kinds are defined? For example can you put numeric indexed parameters before *and* after named ones? eg `Articles::view/param/key:value/param2`?

Should we be concerned that the `key:value` syntax looks the same as 'named parameters' which is a feature we removed in 3.x? Should we use `key=value` so that the syntax is closer to query string parameters? Or is that more confusing?

Reply:
I'll try the `key=value` approach later today

Reply:
> Should we be concerned that the key:value syntax looks the same as 'named parameters' which is a feature we removed in 3.x?

That's a valid concern. Reusing the separator char we deprecated could be confusing for users.

Reply:
The only part I am not sure is which kind of characters we should allow to be part of the `key=value` part.
Currently we restrict it to `[a-z0-9_/:=]`

Reply:
The key regex should be same as https://github.com/cakephp/cakephp/blob/cf67e6148943d9d67025e697ec240a0a91b2720e/src/Routing/Route/Route.php#L116 sans the braces.

Reply:
Wasn't able to just use it directly but I hopeyfully adjusted it now to what it should be ^^